src/BloomExe/WebLibraryIntegration/BloomS3Client.cs

```diff
@@ -355,13 +355,29 @@ public string DownloadFile(string storageKeyOfFile)
 		/// <param name="storageKeyOfBookFolder"></param>
 		public string DownloadBook(string storageKeyOfBookFolder, string pathToDestinationParentDirectory, ProgressDialog downloadProgress = null)
 		{
-			//TODO tell it not to download pdfs. Those are just in there for previewing purposes, we don't need to get them now that we're getting the real thing
-
 			//review: should we instead save to a newly created folder so that we don't have to worry about the
 			//other folder existing already? Todo: add a test for that first.
 
-			if (!GetBookExists(storageKeyOfBookFolder))
+			// We need to download individual files to avoid downloading unwanted files (PDFs and thumbs.db to
```

Comment:
It takes a while to figure out what this block of code is actually trying to accomplish. I *think* it's just trying to compute a total number of items that will get downloaded. COnsider factoring out a method then, something like ```GetTotalItemsThatWillBeDownloaded()```